README.md

```diff
@@ -71,16 +71,17 @@ kind: DataSource | SLO | SLI | AlertPolicy | AlertCondition | AlertNotificationT
 metadata:
   name: string
   displayName: string # optional
-  labels: # optional, key <>value a pair of labels that can be used as metadata relevant to users
-    # Example labels
+  labels: # optional, it's allowed to assign multiple values to a single key
+    # example labels
     userImpacting: "true"
```

Comment:
Is it possible in golang to read a field as either string or string[]? It's the core of the problem I described in the issue.

Until we can prove the validation code can support this use case, we shouldn't update the spec with this information

Reply:
It will always be unmarshaled into `[]string`, I proposed this in https://github.com/OpenSLO/oslo/issues/202. It's doable in our YAML library. So you can evolve this and firstly go with
```yml
...
  labels:
    userImpacting: "true"
...
```
and later change to
```yml
...
  labels:
    userImpacting:
      - "a"
      - "b"
... 
```
and the underlying structure will be ready to handle this

Reply:
If it's always unmarshalled to `string[]`, doesn't that imply a single value needs to be specified as:
```
...
  labels:
    userImpacting:
      - "true"
... 
```

Reply:
Personally find this a bit awkward to define a flag/boolean in such manner. Not well versed in yaml parsing in Golang

Reply:
We can do this in a way that both will work @kenfinnigan here is a snippet for this https://github.com/go-yaml/yaml/issues/100#issuecomment-901604971.

I recommend this way because it won't break anyone existing configurations done with solo.

Good point @weyert. Labels don't support any other type than `string` this is the only example of something that may make sense for a user from the semantic perspective that she wants to filter by it. But maybe an example with `"true"` is not the best, it drives confusion. Hence I'm removing this and replace with another label

Reply:
@programmer04 can this be easily incorporated into Oslo? Last I recall from looking at the code, the unmarshalling of the entire structure was based on the golang structs and we're not walking the tree of nodes unmarshalling as we go

Reply:
We checked this with @fpiwowarczyk today and we think it's achievable

Reply:
@kenfinnigan & @weyert do you think that this conversation can be marked as resolved?

Reply:
I'd prefer to hold this PR for approval until a PR implementing it in Oslo is created and the solution validated. Otherwise there's the risk of modifying the spec, and then modifying it again, for the same change.